src/SURLCreator.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright (c) 2021 Chiu Yen-Chen, Swen Sun-Yen, Wen Yong-Wei.
+All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file. See the AUTHORS file for names of contributors.
+ */
+
+#include "SURLCreator.h"
+
+namespace shortlink {
+
+const std::string SURLCreator::base64_table =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-";
+
+std::string SURLCreator::CreateSURL(const std::string &raw_url) {
+  // get hash Code in length of 20
+  size_t hashCode = Hasher::HashRawUrl(raw_url);
+
+  // transfer hash Code to Hex
+  std::stringstream stream;
+  stream << std::hex << hashCode;
+  std::string result(stream.str());
+
+  // transfer to binary
+  std::string binary = "";
+  for (int i = 0; i < result.size(); i++) {
+    binary += hex2bits(result[i]);
+  }
+
+  // transfer each six bit to base64 char
+  std::string surl_key = "";
+  for (int i = 0; binary.size() - i >= 6; i += 6) {
+    std::string subChar = binary.substr(i, 6);
+    int ids = std::stoi(subChar, 0, 2);
+    surl_key += base64_table[ids];
+  }
+
+  return surl_key;
```

Comment:
Would this be too long? How long do you want your short URLs to be?

Reply:
the length of short url is 8.
total length of surl_key is 10.
So we drop out the last two.